src/stransport_stream.c

```diff
@@ -150,9 +156,9 @@ static OSStatus read_cb(SSLConnectionRef conn, void *data, size_t *len)
 	requested = left = *len;
 	do {
 		ret = git_stream_read(io, data + (requested - left), left);
-		if (ret < 0) {
+		if (ret <= 0) { /* There was an error or nothing could be read because stream is closed */
 			*len = 0;
-			return -1;
+			return errSSLClosedAbort;
```

Comment:
No matter what, all these uses of the API do loop for the write callback, so that's one thing to fix in libgit2.